### DISCUSSION TITLE
How to execute functions from a NodeJs project in the Quasar backend ?

### DISCUSSION THREAD

**Discussion by Spixz** [quasarframework/quasar#14762] [General - Components / Directives / etc] [answered]
Hello everyone, I am new to Quasar and just finished the Quick Start guide. I would like to create a desktop app. So I installed quasar-cli and created my project.

I would like to **import functions from a NodeJS project** (OCR, pdf generation, ...) **using different modules** and **run them in the back-end of my Quasar application** and not in the front-end.
I saw on the [quasar forum](https://forum.quasar-framework.org/topic/6016/where-do-i-add-3rd-party-library-functions-to-execute-on-app-load) that people use Boot File but I checked the documentation and it's not advised. From what I understand, boot files are used to do configuration.
According to [this discussion](https://github.com/quasarframework/quasar/discussions/14744), I place a test function in the _/src_ folder and imported it in my Vue component like explained. It worked but the code run in the front-end.

So my question is this: **How to execute functions from a NodeJs project in the quasar backend and have access to the results in the front end?**

Thanks and have a nice day ;)

**Comment by MilosPaunovic** [comment, marked as answer]
Are you building an Electron app?

**Reply by Spixz** [reply]
Ah, I thought Quasar was using its own system.

![image](https://user-images.githubusercontent.com/43412722/199194279-a2997ae1-9214-4369-9f17-58230ec358fd.png)

I have just seen that you have to add electron to Quasar and I have found the necessary [documentation.](https://quasar.dev/quasar-cli-vite/developing-electron-apps)
I should then use the script [preloading system](https://quasar.dev/quasar-cli-vite/developing-electron-apps/electron-preload-script).
Thanks a lot for your answer.
